v2/car_test.go

```diff
@@ -74,5 +75,109 @@ func TestEmptyCharacteristics(t *testing.T) {
 }
 
 func TestHeader_SizeIs40Bytes(t *testing.T) {
-	assert.Equal(t, 40, new(car_v2.Header).Size())
+	assert.Equal(t, car_v2.HeaderBytesSize, new(car_v2.Header).Size())
+}
+
+func TestHeader_Marshal(t *testing.T) {
+	tests := []struct {
+		name        string
+		target      car_v2.Header
+		wantMarshal []byte
+		wantErr     bool
+	}{
+		{
+			"header with empty characteristics is marshalled as expected",
```

Comment:
nit: test and sub-test names as printed by `go test` can't contain spaces, so those will get converted to underscores.

that will also look somewhat inconsistent, like `TestHeader_Marshal/header_with_empty_characteristics_is_marshalled_as_expected`.

IMO that's also a bit confusing, as underscores in places like `TestHeader_Marshal` mean a separation, not simply splitting words.

I personally use camel case for subtest names, so it would look like:

> TestHeader_Marshal/HeaderWithEmptyCharacteristicsIsMarshalledAsExpected

Reply:
Thanks for pointing that out. That's a reasonable approach. 

My take is that CamelCase is harder to read than simply spaces for humans whom look at the code. But then it keeps the test name and `go test` output consistent.

The `_` in the test method name signifies a receiver function defined for type `Header`.  I used to call the function `Marshal`; It is now called `WriteTo` so I'll fix that. 

Is the CamelCase the convention generally followed in IPLD? Happy to name tests in that convention if that keeps things consistent.

Reply:
Right, consistency is what I'm after here. There's the consistency between the code and the output, but also the consistency that "_" is often a separator between type and method, or a separator between a name and a description, like `ExampleWriter_veryLarge`.

I don't think there's a hard convention anywhere about this, actually. The official docs happen to not use multiple words, only subtest names like Test1. And blog posts like https://blog.golang.org/subtests actually use a mix of both, which I find pretty funny.

So I guess follow what's consistent with the rest of go-car :) I still lean towards camel case by default, but I can't say it's the one true way.

Reply:
There is no current strong convention in go-car. Let's make the CamelCase it :) Thanks @mvdan